src/cursor/cur_file.c

```diff
@@ -1121,6 +1121,10 @@ __wt_curfile_open(WT_SESSION_IMPL *session, const char *uri, WT_CURSOR *owner, c
             WT_RET_MSG(session, EINVAL, "Value for 'bulk' must be a boolean or 'bitmap'");
 
         if (bulk) {
+            /* Bulk cursors can't be opened inside transactions. */
+            if (F_ISSET(session->txn, WT_TXN_RUNNING))
+                WT_RET(EINVAL);
```

Comment:
Could you turn this into WT_RET_MSG, and add some text explaining why it is an error.

Reply:
Good suggestion! 
I've switch to WT_RET_MSG, added a message, and removed the now redundant comment (as it effectively just duplicates the new error message).